ui/pages/confirmation/templates/add-ethereum-chain.js

```diff
@@ -385,7 +385,21 @@ function getValues(pendingApproval, t, actions, history) {
         pendingApproval.requestData,
       );
       if (originIsMetaMask) {
-        actions.addCustomNetwork(pendingApproval.requestData);
+        const networkConfigurationId = await actions.upsertNetworkConfiguration(
+          {
+            ...pendingApproval.requestData,
+            nickname: pendingApproval.requestData.chainName,
+          },
+          {
+            source: pendingApproval.requestData.source,
+            referrer: pendingApproval.requestData.referrer,
```

Comment:
This is hard-coded in the action being called here (luckily this block is only for internally-triggered additions, so the metamask origin is correct to use here)
```suggestion
```